TCD-2892: Backtrace (#2145)
saucelabs/sauce-docs: docs/error-reporting/platform-integrations/visual-studio.md

- The submission URL field provides an autocomplete feature based on the instance name you enter.
- The submission token field to allow crashes to be submitted to the Backtrace instance. This field requires that the provided string has exactly 64 characters. If you don't have a Backtrace instance, you can create one by following the provided link.
- Don't have an account?: This link will take the user to a web page where they can create a new Backtrace instance. Use this link if you don't already have a Backtrace instance provisioned for your project.
- Click **Next** to proceed to the next page.
  <img src={useBaseUrl('/img/error-reporting/vs-extension/5e601bc4c83c1.png')} alt="step 1: backtrace server information"/>

**Step 2: Download or Use Existing Crashpad Library**

- This step allows you to choose the versions of Crashpad you want to download (from the get.backtrace.io/crashpad/builds page) or specify the directory of an already downloaded library.
- Select the desired configuration and platform versions from the dropdown lists. Backtrace provides both Release and Debug configurations of Crashpad, and x64 & Win32 platforms.

> - Use the **Download Crashpad** to connect to the `get.backtrace.io` servers and download the specified versions.You should select the destination directory, and the wizard will download the latest Crashpad binaries. If you specified multiple Configurations or Platforms to download, the wizard will download each of those.

Author: ```suggestion
- Use the **Download Crashpad** to connect to the `get.backtrace.io` servers and download the specified versions. You should select the destination directory, and the wizard will download the latest Crashpad binaries. If you specify multiple Configurations or Platforms to download, the wizard will download each.
```

</ul>

You may see dialogs notifying you that files are being modified outside the Visual Studio environment. Choose **Reload All** as necessary.
<img src={useBaseUrl('/img/error-reporting/vs-extension/5e601bc9156b3.png')} alt="Reload all button"/>

A dialog about inconsistent line endings may appear. It's recommended to normalize the line endings according to your environment.
<img src={useBaseUrl('/img/error-reporting/vs-extension/5e601bca1b4c4.png')} alt="inconsistent line endings dialog"/>

The Visual Studio Extension modifies the vcxproj file (project definition file) to add Crashpad to your project. You can further customize each option in **Project Properties**.

If you enable symbol upload, a post-build task is created. After each successful build, the extension sends the necessary files (.PDBs, .DLLs, .EXEs) as a zip archive to the Backtrace instance. You can disable this option in **Project Properties** under **Post-build** actions.

## Post Install Steps

After completing the wizard, make sure to open the CrashpadSetup.cpp file and specify the following values:

- **db_path**: Set the directory where crash reports should be locally stored before submitting to Backtrace.

> - **handler_path**: Configure the launch mechanism for Crashpad (e.g., out-of-process).

Author: ```suggestion
- **handler_path**: Configure the launch mechanism for Crashpad (for example., out-of-process).
```
